limit_checker.py

```diff
@@ -0,0 +1,25 @@
+from logger import Logger
+class LimitChecker:
+    def __init__(self) -> None:
+        self.logger = Logger()
+        self.value = None
+    
+    def is_in_range(self, min, max):
+        result = self.value < min or self.value < max
+        self.logger.log_range_output(result, self.param_name)
+        return result
+
+    def is_in_high_range(self, max):
+        result = self.value < max
+        self.logger.log_higher_output(result, self.param_name)
+        return result
+
+    def is_in_low_range(self, min):
+        result = self.value > min
+        self.logger.log_lower_output(result, self.param_name)
+        return result
+
+    def check_param(self, value, param_name, min = 0, max = 0):
+        self.value = value
+        self.param_name = param_name
+        return all([self.is_in_range(min, max), self.is_in_high_range(max), self.is_in_low_range(min)])
```

Comment:
Does this code execute all range functions? if it executes all 3 then optimization can be done here.i.e. if it is in range then other 2 need not be executed! Please check

Reply:
Yes, I have modified as appropriate